File: 1529_BulbSwitcherIV.py
class Solution:
    def minFlips(self, target: str) -> int:
        initial = "0"
        output = 0
        for i in target:
            if i != initial:
                output += 1
                initial = "1" if initial == "0" else "0"
        return output
    
    ## Time Complexity = O(n)
    ## Space Complexity = O(1)

File: C++/1529_BulbSwitcherIV.cpp
class Solution {
public:
    int minFlips(string target) {
        char initial = '0';
        int output = 0;
        for (int i = 0; i < target.size(); i++){
            if (target[i] != initial){
                output++;
                initial = initial == '1' ? '0' : '1';
            }
        }
        return output;
    }
};